packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -101,6 +107,11 @@ export class DockerImageAsset extends Construct implements assets.IAsset {
     if (props.file)           { extraHash.file = props.file; }
     if (props.repositoryName) { extraHash.repositoryName = props.repositoryName; }
 
+    // add "salt" to the hash in order to invalidate the image in the upgrade to
+    // 1.21.0 which removes the AdoptedRepository resource (and will cause the
+    // deletion of the ECR repository the app used).
+    extraHash.version = '1.21.0';
```

Comment:
Guess it doesn't really matter what this is, but if you want it to mean the version this was introduced, it should now read `1.22.0`.

Reply:
I left it "1.21.0" because it would require running all integ tests *again*.